sources/api/api-exec.md

```diff
@@ -0,0 +1,106 @@
+
+### Capacity updates
+
+Actix gives us a great way to communicate with the client over a WebSocket, but seemingly no way to control the receipt of the messages.
+In a manual server implementation, you can control when you pull data from a network socket, and you can wait to pull more until you've processed the previous data.
+In the Actor-based WebSocket implementation, you receive a stream of messages, and data is still being received as you process each message of the stream.
+This can be a problem when the client needs to send a lot of data for processing; the buffers will keep filling up and can overload the server.
+
+The typical solution to this is "backpressure," which means giving some kind of signal to the sender that they're overloading the channel.
+Rust has bounded channels that accomplish this - notifying the sender that the channel is full - and indeed we use one to hold input from the user.
+The trick here is that we don't control the input buffer, so we can't just "naturally" block network traffic - Actix controls that.
+
+We fix this by sending regular capacity updates to the client.
+Messages are processed in order, and the client knows how many chunks of input it's read from the user, so if we tell them how many of their messages we've finished writing to the process, and the maximum number of messages we'll allow outstanding, they can throttle themselves.
+For example, if we allow 1,024 outstanding messages, the client has sent 1,024, and the server has written 0, the client knows it has to wait until some are written.
+(If they don't, we terminate the connection.)
+
+The client implements this by updating an `AtomicCapacity` in `ReadFromServer`, and waiting for there to be available capacity before reading more user input in `ReadFromUser`.
+(`AtomicCapacity` just stores the numbers we get from the server, and can be shared across those two threads.)
+
+We don't have to worry about capacity in the opposite direction because (1) the client only reads another message from the WebSocket stream when it's written the prior one to stdout, and (2) writing output on the client side is generally faster than the server plus the network.
+
+### Terminal management
+
+#### Terminal settings
+
+If you start an interactive program from a shell, and you press ctrl-c, you want the program to stop so you can be returned to the shell.
```

Comment:
I guess I've seen the following question enough that I default to lowercase to avoid ambiguity :)  "Does Ctrl-C mean capital C?  Do I need to hold shift??"